metrics_sdk/lib/opentelemetry/sdk/metrics/export/periodic_metric_reader.rb

```diff
@@ -123,7 +123,18 @@ def start
           def export(timeout: nil)
             @export_mutex.synchronize do
               collected_metrics = collect
-              @exporter.export(collected_metrics, timeout: timeout || @export_timeout) unless collected_metrics.empty?
+              result_code = @exporter.export(collected_metrics, timeout: timeout || @export_timeout) unless collected_metrics.empty?
+              report_result(result_code)
```

Comment:
I think there's a bug: when `collected_metrics` is an empty array, `report_result` will log an error because `result_code` is `nil`

Reply:
Thanks! LGTM